File: Example/JavaClass/Model/com/github/hebertialmeida/modelgen/models/Company.java
package com.github.hebertialmeida.modelgen.models;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import android.net.Uri;

/// Definition of a Company
public final class Company {

    final int id;
    final @Nullable Uri logo;
    final @NonNull String name;
    final @NonNull String subdomain;

    public Company(
            int id,
            @Nullable Uri logo,
            @NonNull String name,
            @NonNull String subdomain
            ) {

        this.id = id;
        this.logo = logo;
        this.name = name;
        this.subdomain = subdomain;
    }

    public int getId() {
        return id;
    }

    @Nullable
    public Uri getLogo() {
        return logo;
    }

    @NonNull
    public String getName() {
        return name;
    }

    @NonNull
    public String getSubdomain() {
        return subdomain;
    }

}

File: Example/DataClass/Model/com/github/hebertialmeida/modelgen/models/Company.kt
package com.github.hebertialmeida.modelgen.models

import android.os.Parcelable
import kotlinx.android.parcel.Parcelize
import android.net.Uri

/// Definition of a Company
@Parcelize
data class Company(
    val id: Int,
    val logo: Uri?,
    val name: String,
    val subdomain: String
): Parcelable
